### ISSUE TITLE
🐛 BUG: <select> element with "mapped" data breaks following markup in `.astro` files

### ISSUE THREAD

**Issue by edburtnieks** [withastro/astro#3105]
### What version of `astro` are you using?

1.0.0-beta.10

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

npm

### What operating system are you using?

Linux

### Describe the Bug

Everything happens in `.astro` files

When I add native HTML `<select>` element with hard coded options, everything works fine. But when I try to map through the options imported from frontmatter part, all the markup after that (e.g. text fields, buttons etc.) gets turned in plain text nodes.

---

* **Note 1:** In the same scenario with `<ul>` and `<li>` elements, everything works as expected
* **Note 2:** The current workaround I did was to export only the `<select>` in it's own `.astro` component and use the component in `<form>` instead

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-f5m6mf?file=src/pages/index.astro

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by itsmattsoria** [comment]
@edburtnieks I *just* experienced this same issue earlier today! I also created a reporducible example and shared it the Discord, but haven't heard anything on it yet: https://stackblitz.com/edit/github-dnzhtk?file=src/components/Tour.astro

I'm brand new to Astro so I've been going in circles for a couple hours trying to determine that I wasn't just doing something wrong, but it does seem like a bug, doesn't it?

**Comment by edburtnieks** [comment]
@itsmattsoria Yea, for few minutes I was questioning myself do I even know basic HTML anymore :sweat_smile: 

But then tried to put it on it's own separate `.astro` component and it worked. I would consider it a bug, because I'm 99% my HTML markup is correct.

**Comment by itsmattsoria** [comment]
For added context: it's not just mapped data, it's any astro vars. I tested out a select field with a single option using a string set as a var in the frontmatter and it had the same issue. Same story — switch that select to any other element and no issue.

**Comment by natemoo-re** [comment]
Yep, this is definitely a bug. I'm going to transfer this to the compiler repo to track.
